src/util/server.ml

```diff
@@ -227,6 +240,27 @@ let () =
     let process () _ = Preprocessor.dependencies_to_yojson ()
   end);
 
+  register (module struct
+    let name = "functions"
+    type params = unit [@@deriving of_yojson]
+    type response = Function.t list [@@deriving to_yojson]
+    let process () serv = Function.getFunctionsList serv.file.globals
+  end);
+
+  register (module struct
+    let name = "cfgs"
+    type params = { fname: string }  [@@deriving of_yojson]
+    type response = { cfg : string } [@@deriving to_yojson]
+    let process { fname } serv = 
+      let (module CFG) = Control.compute_cfg serv.file in 
```

Comment:
This recreates the CFGs for all functions every time a single function is requested, which is wasteful for bigger programs.
You can cherry-pick and use 692eb844b4feb6fb05a255bf1dd379e93962c11b.